src/cycling/miscHandling.java

```diff
@@ -1,5 +1,177 @@
 package cycling;
 
-public class miscHandling {
+import java.util.HashMap;
+
+public class MiscHandling {
     
+    // Hash of team IDs to their respective team object
+    private HashMap<Integer, Team> teamsHash;
+    // Hash of race IDs to their respective race object
+    private HashMap<Integer, Race> racesHash;
+
+    /**
+     * Adds a team object linked to its ID into a team hashmap
+     * 
+     * @param teamObject The object to be added to the array
+     * 
+     * @throws DuplicatedResultException If the object has already been added
+     */
+    public void add(Team teamObject) throws DuplicatedResultException {
+
+        Integer teamID = teamObject.getTeamID();
+
+        if (teamsHash.put(teamID, teamObject) != null) {
+            throw new DuplicatedResultException("This team has already been added");
+        }
+    }
+
+    /**
+     * Adds a race object linked to its ID into a race hashmap
+     * 
+     * @param raceObject The object to be added to the array
+     * 
+     * @throws DuplicatedResultException If the object has already been added
+     */
+    public void add(Race raceObject) throws DuplicatedResultException{
+
+        Integer raceID = raceObject.getRaceID();
```

Comment:
Using Integer wrapper type, not int here

Reply:
yeah, since it then goes into the hashmap and the hashmap uses integers